cpu/cc2538/Makefile.include

```diff
@@ -1,26 +1,44 @@
 
 # Set ROM and RAM lengths according to CPU model
 ifneq (,$(filter cc2538nf11,$(CPU_MODEL)))
-  ROM_LEN ?= 128K
-  RAM_LEN ?= 16K
+  ROM_LEN ?= 0X20000 # 128K
+  RAM_LEN ?= 0x4000  # 16K
 endif
 ifneq (,$(filter cc2538nf23,$(CPU_MODEL)))
-  ROM_LEN ?= 256K
-  RAM_LEN ?= 32K
+  ROM_LEN ?= 0X40000 # 256K
+  RAM_LEN ?= 0x4000  # 32K
 endif
 ifneq (,$(filter cc2538nf53 cc2538sf53,$(CPU_MODEL)))
-  ROM_LEN ?= 512K
-  RAM_LEN ?= 32K
+  ROM_LEN ?= 0X80000 # 512K
+  RAM_LEN ?= 0x4000  # 32K
 endif
 
+# If using riotboot loose 2 pages to not write over CCA section and
+# keep page parity (so slots are split evenly among pages)
+ifneq (,$(filter riotboot,$(FEATURES_USED)))
+  CCA_SIZE ?= 0x800
+else
+  CCA_SIZE ?= 0
+endif
+
+# Set ROM and RAM start address
 ROM_START_ADDR ?= 0x00200000
 RAM_START_ADDR ?= 0x20000000
 
-KB := 1024
-ROM_LEN_K := $(shell echo $(ROM_LEN) | sed 's/K//')
-FLASHSIZE := $(shell echo $$(( $(ROM_LEN_K) * $(KB) )) )
-
 # Set CFLAGS
+FLASHSIZE := $(shell echo $$(( $(ROM_LEN) - 2 * $(CCA_SIZE) )) )
 CFLAGS += -DCC2538_FLASHSIZE=$(FLASHSIZE)U
 
+# Use common ld script
+LINKER_SCRIPT ?= cc2538.ld
```

Comment:
Have those linker scripts in `cc2538/ldscripts` been used before? I could not find any reference.

Reply:
By default it seems to be set in `cortexm.inc.mk` to `LINKER_SCRIPT ?= $(CPU_MODEL).ld`